### PR TITLE
added resourcegroupstaggingapi support for ssm

### PR DESCRIPTION
**Pull request by vgiridhar97**

### CODE REVIEW

File: tests/test_resourcegroupstaggingapi/test_resourcegroupstaggingapi.py [hunk 1]
```diff
@@ -759,6 +759,32 @@ def test_get_resources_sns():
         "Tags"
     ]
 
+@mock_aws
+def test_get_resources_ssm():
+    import json
+    import pkgutil
+    import yaml
+
+    template_file = pkgutil.get_data(__name__, "test_templates/good.yaml")
```
Comment (bblommers) [inline tests/test_resourcegroupstaggingapi/test_resourcegroupstaggingapi.py:768]: Can we reuse this from the SSM tests?

I think we can just do a toplevel import:
`from tests.test_ssm/test_ssm_docs import _get_yaml_template`

That way we don't have to duplicate anything

File: moto/resourcegroupstaggingapi/models.py [hunk 2]
```diff
@@ -509,6 +513,23 @@ def format_tag_keys(
...
+        # SSM 
+        if not resource_type_filters or "ssm" in resource_type_filters:
+            for document in self.ssm_backend._documents.values():
+                doc_name = document.describe()["Name"]
+                tags = self.ssm_backend._get_documents_tags(doc_name)
+                if not tags or not tag_filter(
+                    tags
+                ): # Skip if no tags, or invalid filter
+                    continue
+                # pytest.set_trace()
```
Comment (bblommers) [inline moto/resourcegroupstaggingapi/models.py:526]: This should be removed

File: moto/ssm/models.py [hunk 3]
```diff
@@ -1286,6 +1287,7 @@ def create_document(
         self._documents[ssm_document.name] = documents
 
         if tags:
+            # pytest.set_trace()
```
Comment (bblommers) [inline moto/ssm/models.py:1290]: Same here (and the `pytest`-import)

**Review by bblommers — CHANGES_REQUESTED** [review]
Hi @vgiridhar97! Just a few minor points - the change itself looks good to me.

File: moto/resourcegroupstaggingapi/models.py (hunk 2 above)
Comment (vgiridhar97) [reply moto/resourcegroupstaggingapi/models.py]: hi, i've removed this comment in my latest commit

File: moto/ssm/models.py (hunk 3 above)
Comment (vgiridhar97) [reply moto/ssm/models.py]: hi, i've removed this comment in my latest commit

File: tests/test_resourcegroupstaggingapi/test_resourcegroupstaggingapi.py (hunk 1 above)
Comment (vgiridhar97) [reply tests/test_resourcegroupstaggingapi/test_resourcegroupstaggingapi.py]: hi, i've changed it to the toplevel import in my latest commit

**Review by bblommers — CHANGES_REQUESTED** [review]
Nitpick - but the additional yaml-file is no longer necessary now.

Looks like the linter is failing as well: https://github.com/getmoto/moto/actions/runs/8254430099/job/22578588665

File: moto/ssm/models.py [hunk 4]
```diff
@@ -8,6 +8,7 @@
 from typing import Any, DefaultDict, Dict, Iterator, List, Optional, Tuple
 
 import yaml
+import pytest
```
Comment (bblommers) [inline moto/ssm/models.py:11]:
```suggestion
```

**Review by bblommers — APPROVED** [review]
LGTM - thank you for adding this @vgiridhar97!

**Pull request merged**